writer.go

```diff
@@ -0,0 +1,119 @@
+package pool
+
+import (
+	"bufio"
+	"io"
+	"sync"
+)
+
+const WriterBufferSize = 4096
+
+var bufioWriterPool = sync.Pool{
+	New: func() interface{} {
+		return bufio.NewWriterSize(nil, WriterBufferSize)
+	},
+}
+
+// Writer is a buffered writer that returns its internal buffer in a pool when
+// not in use.
+type Writer struct {
```

Comment:
don't we also need a constructor?

Reply:
Not really, I'm treating this like https://golang.org/pkg/io/#LimitedReader. Although, that _will_ make it a bit tricky to introduce different sized buffers. I don't have strong opinions on this (my main goal was to avoid an allocation.

Reply:
ok, fair enough.